### DISCUSSION TITLE
Allow icons in code snippet headers

### DISCUSSION THREAD

**Discussion by tmkn** [withastro/starlight#1854] [Feature Requests]
### What version of `starlight` are you using?

0.22.2

### What is your idea?

When you give your code snippets a title, they will be displayed like this:
![localhost_4321_javascript-patterns_array-concatenate_](https://github.com/withastro/starlight/assets/2671613/f24a9b08-fa6f-4e5b-aea2-5dc874d81cc4)

Imo it would be nice if one could specify an icon to be displayed before the title, like the tab component does.

Something like
```
    ```typescript title="TypeScript" icon="seti:typescript" <- use already shipped icons

    ```
```

### Why is this feature necessary?

I think it would be a nice feature, especially if you work on a site that shows code for different languages, this would make it a little easier to identify which code is which.

Further, Starlight already comes with a lot of programming icons, why not use them for something like this :)

### Do you have examples of this feature in other projects?

_No response_

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thank you for the suggestion @tmkn — I think it could indeed be a nice feature.

Starlight uses [Expressive Code](https://expressive-code.com/) to render code blocks, so this might need a feature request upstream there.

**Reply by HiDeoo** [reply]
An expressive code plugin is [now available](https://github.com/xt0rted/expressive-code-file-icons) for this feature and a walkthrough of how to use it is available [here](https://hackingonhacks.com/blog/2025/02/13/adding-file-icons-to-expressive-code-frames).

**Comment by HiDeoo** [comment]
As this feature is now available as an Expressive Code plugin, I'm going to close this discussion.
